pkg/ddevapp/ddevapp.go

```diff
@@ -976,12 +976,20 @@ Fix with 'ddev config global --required-docker-compose-version="" --use-docker-c
 
 	if app.IsMutagenEnabled() {
 		if ok, volumeExists, info := CheckMutagenVolumeSyncCompatibility(app); !ok {
-			util.Debug("mutagen sync session and docker volume are in incompatible status: '%s', Removing mutagen sync session '%s' and docker volume %s", info, MutagenSyncName(app.Name), GetMutagenVolumeName(app))
+			util.Debug("mutagen sync session, configuration, and docker volume are in incompatible status: '%s', Removing mutagen sync session '%s' and docker volume %s", info, MutagenSyncName(app.Name), GetMutagenVolumeName(app))
 			terminateErr := TerminateMutagenSync(app)
```

Comment:
I think my only comment would be that it might be worth flushing the stale session before terminating it, that way both endpoints are theoretically the same when the new session is created.  Doing that will help ensure that the first reconciliation pass of the new synchronization session will construct a virtual ancestor that's (ideally) equal to both endpoints (and that way no pending unsynced changes are lost and converted to conflicts when the new empty ancestor is created in the new session).

However, my guess is that a flush may be done elsewhere, just before `CheckMutagenVolumeSyncCompatibility` (and potentially `TerminateMutagenSync`) is invoked, so if that's the case then just ignore this comment.

Mutagen should probably eventually add a mechanism to transfer the virtual ancestor from one session to a new one (maybe a `mutagen sync update` command, or something).  There are some complexities with that, but it should be possible someday.

Reply:
Thanks so much, will check!